packages/tailwindcss/src/compat/apply-compat-hooks.ts

```diff
@@ -142,38 +147,48 @@ export async function applyCompatibilityHooks({
   // any additional backwards compatibility hooks.
   if (!pluginPaths.length && !configPaths.length) return
 
-  let configs = await Promise.all(
-    configPaths.map(async (configPath) => ({
-      path: configPath,
-      config: await loadConfig(configPath),
-    })),
-  )
-  let pluginDetails = await Promise.all(
-    pluginPaths.map(async ([pluginPath, pluginOptions]) => ({
-      path: pluginPath,
-      plugin: await loadPlugin(pluginPath),
-      options: pluginOptions,
-    })),
-  )
+  let [configs, pluginDetails] = await Promise.all([
+    Promise.all(
+      configPaths.map(async ({ id, base }) => {
+        let loaded = await loadModule(id, base, 'config')
+        return {
+          path: id,
+          base: loaded.base,
+          config: loaded.module as UserConfig,
+        }
+      }),
+    ),
+    Promise.all(
+      pluginPaths.map(async ([{ id, base }, pluginOptions]) => {
+        let loaded = await loadModule(id, base, 'plugin')
+        return {
+          path: id,
+          base: loaded.base,
+          plugin: loaded.module as Plugin,
+          options: pluginOptions,
+        }
+      }),
+    ),
+  ])
 
-  let plugins = pluginDetails.map((detail) => {
+  let pluginConfigs = pluginDetails.map((detail) => {
     if (!detail.options) {
-      return detail.plugin
+      return { config: { plugins: [detail.plugin] }, base: detail.base }
     }
 
     if ('__isOptionsFunction' in detail.plugin) {
-      return detail.plugin(detail.options)
+      return { config: { plugins: [detail.plugin(detail.options)] }, base: detail.base }
     }
 
     throw new Error(`The plugin "${detail.path}" does not accept options`)
   })
 
-  let userConfig = [{ config: { plugins } }, ...configs]
+  let userConfig = [...pluginConfigs, ...configs]
 
   let resolvedConfig = resolveConfig(designSystem, [
-    { config: createCompatConfig(designSystem.theme) },
+    { config: createCompatConfig(designSystem.theme), base: '' },
     ...userConfig,
-    { config: { plugins: [darkModePlugin] } },
+    { config: { plugins: [darkModePlugin] }, base: '' },
```

Comment:
Maybe use the `base` variable here instead of an empty string? We don't actually use the empty string for anything now but if we ever did have to reference base for another thing down the road at least then things would work.